types/algoliasearch/index.d.ts

```diff
@@ -1880,9 +1881,254 @@ declare namespace algoliasearch {
+        /**
+         * Restrict the search results to a specific type.
+         * https://community.algolia.com/places/api-clients.html#api-options-type
+         */
+        hitsPerPage?: number;
+        /**
+         * If specified, restrict the search results to a single language. You can pass two letters country codes (ISO 639-1).
+         * Warning: language parameter is case sensitive and should be lowercase otherwise it will fallback to default language.
+         * https://community.algolia.com/places/api-clients.html#api-options-language
+         */
+        language?: string;
+        /**
+         * If specified, restrict the search results to a specific list of comma-separated countries. You can pass two letters country codes (ISO 3166-1).
+         * Default: Search on the whole planet.
+         * Warning: country codes must be lower-cased.
+         * https://community.algolia.com/places/api-clients.html#api-options-countries
+         */
+        countries?: string;
+        /**
+         * Force to first search around a specific latitude longitude.
+         * The option value must be provided as a string: latitude,longitude like 12.232,23.1.
+         * The default is to search around the location of the user determined via his IP address (geoip).
+         * https://community.algolia.com/places/api-clients.html#api-options-aroundLatLng
+         */
+        aroundLatLng?: string;
+        /**
+         * Whether or not to first search around the geolocation of the user found via his IP address. This is true by default.
+         * https://community.algolia.com/places/api-clients.html#api-options-aroundLatLngViaIP
+         */
+        aroundLatLngViaIP?: string;
+        /**
+         * Radius in meters to search around the latitude/longitude. Otherwise a default radius is automatically computed given the area density.
+         * https://community.algolia.com/places/api-clients.html#api-options-aroundRadius
+         */
+        aroundRadius?: number;
+        /**
+         * Controls whether the _rankingInfo object should be included in the hits. This defaults to false.
+         * The _rankingInfo object for a Places query is slightly different from a regular Algolia query
+         * and you can read up more about the difference and how to leverage them in our guide.
+         * https://community.algolia.com/places/api-clients.html#api-options-getRankingInfo
+         */
+        getRankingInfo?: boolean;
+    }
+
+    interface PlaceInterface {
+        /**
+         * Endpoint to search.
+         * https://community.algolia.com/places/api-clients.html#endpoints
+         */
+        search(e: QueryInterface, cb: (err: Error, response: ResultSearchInterface) => void): void;
+
+        /**
+         * Endpoint to search.
+         * https://community.algolia.com/places/api-clients.html#endpoints
+         */
+        search(e: QueryInterface): Promise<ResultSearchInterface>;
```

Comment:
See my comments about the `language` parameter further below.

I don't know if this is valid typescript but we should probably have two separate declarations for the search method, if it's possible:

```ts
  search(e: QueryInterface): Promise<ResultSearchInterface>;
  search(e: LocalizedQueryInterface): Promise<LocalizedResultSearchInterface>;
```

Reply:
Like Result has same structure if language is set but `hits` is changing, I did that :
```
hits: HitInterface[] | LocalizedHitInterface[];
```